bin/backup.sh

```diff
@@ -0,0 +1,64 @@
+#!/bin/bash
+#######################################
+#  Perform a full backup of the site
+#  on the users home.
+#  This script can be called directly
+#  or automated by cron, as it do a
+#  cleanup for oldest backups.
+#
+#             WARNING!
+#  Execute from the project root
+#######################################
+
+set -x
+
+BACKUPS_DIR_SITE=~/backups-automated/site
+BACKUPS_DIR_DB=~/backups-automated/db
+N_BACKUPS_KEEP=5
+BACKUP_PREFIX=${1:-site}
+
+#######################################
+#  Ensure directory exist
+#
+#  params:
+#    $1: directory absolute path
+#######################################
+function ensure_directory() {
+  if [ ! -d $1 ]
+  then
+    mkdir -p $1;
+  fi
+}
+
+#######################################
+#  Do a cleanup for oldest files
+#
+#  params:
+#    $1: directory absolute path
+#    $2: max number of files to keep
+#######################################
+function cleanup() {
+  cd $1;
+  N_BACKUPS=`ls -1 | wc -l`
+  if [ ${N_BACKUPS} -gt $2 ]
+  then
+    LATEST_BACKUPS=`ls -ct1 | head -n $2 | tr '\n' '|'`;
+    ls -1 | grep -E -v "^(${LATEST_BACKUPS})$" | xargs rm
+  fi
+}
+
+# Ensure directories exist
+ensure_directory ${BACKUPS_DIR_SITE}
+ensure_directory ${BACKUPS_DIR_DB}
+
+# Do the backups
+STAMP=`date +%y%m%d-%H%M%S`;
+CURRENT_DIR=$(basename `pwd`)
+drush sql-dump --gzip > ${BACKUPS_DIR_DB}/${BACKUP_PREFIX}-db-${STAMP}.sql-gz;
```

Comment:
I would add a check to make sure this is a drupal folder.

This how the fix permissions script does the trick:
```
if [ -z "${drupal_path}" ] || [ ! -d "${drupal_path}/sites" ] || [ ! -f "${drupal_path}/core/modules/system/system.module" ] && [ ! -f "${drupal_path}/modules/system/system.module" ]; then
  printf "**********************************************\n"
  printf "* Error: Please provide a valid Drupal path. *\n"
  printf "**********************************************\n"
  print_help
  exit 1
fi
```
See  https://www.drupal.org/node/244924#script-based-on-guidelines-given-above